functions/pollQueue.js

```diff
@@ -0,0 +1,139 @@
+"use strict";
+
+const AWS = require("aws-sdk");
+const DBD = new AWS.DynamoDB.DocumentClient();
+const SQS = new AWS.SQS({ apiVersion: "2012-11-05" });
+const Lambda = new AWS.Lambda({ apiVersion: "2015-03-31" });
+
+const {
+  CONFIG_TABLE,
+  QUEUE_NAME,
+  CONTENT_BUCKET,
+  PROCESS_QUEUE_FUNCTION
+} = process.env;
+
+const RATE_LIMIT = 5;
+const RATE_PERIOD = 1000;
+const MAX_LONG_POLL_PERIOD = 20;
+const POLL_DELAY = 100;
+const EXECUTION_MUTEX_KEY = "pollQueueExecutionExpires";
+const EXECUTION_MUTEX_TTL = 50 * 1000;
+
+const wait = delay => new Promise(resolve => setTimeout(resolve, delay));
+
+// Running list of timestamps for hits on rate limit
+let rateHits = [];
+
+module.exports.handler = async function(event, context) {
+  const now = Date.now();
+
+  try {
+    await DBD.put({
+      TableName: CONFIG_TABLE,
+      Item: {
+        key: EXECUTION_MUTEX_KEY,
+        value: now + EXECUTION_MUTEX_TTL
+      },
+      ConditionExpression: "#key <> :key OR (#key = :key AND #value < :value)",
+      ExpressionAttributeNames: { "#key": "key", "#value": "value" },
+      ExpressionAttributeValues: {
+        ":key": EXECUTION_MUTEX_KEY,
+        ":value": Date.now()
+      }
+    }).promise();  
+  } catch (err) {
+    console.warn("Could not acquire execution mutex", err);
+    return;
+  }
+  console.info("Execution mutex acquired");
+
+  let polls = 0;
+  console.log('Poller start');
+  do {
+    try {
+      const tname = `pollQueue ${++polls}`;
+      console.time(tname);
+      await pollQueue(context);
+      console.timeEnd(tname);
+    } catch (err) {
+      console.error("Error in pollQueue", err);
+    }
+    await wait(POLL_DELAY);
+    console.log("Remaining", context.getRemainingTimeInMillis(), "ms");
+  } while (
+    Math.floor(context.getRemainingTimeInMillis() / 1000) > 1
+  )
+  console.log('Poller exit');
+
+  try {
+    await DBD.delete({
+      TableName: CONFIG_TABLE,
+      Key: { key: EXECUTION_MUTEX_KEY },
+    }).promise();  
+  } catch (err) {
+    console.warn("Could not release execution mutex", err);

```

Comment:
nit: Should we add a `return` here? Otherwise it looks like we could potentially log "Could not release execution mutex..." immediately followed by "Execution mutex released".